File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TDBXBcdValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * 
 * Wrap a binary-compressed decimal values and allows it to be null.
 *
 */

public class TDBXBcdValue extends DBXValue {
	protected boolean ValueNull = false;	

	public TDBXBcdValue() {
		super();
		setDBXType(DBXDataTypes.BcdType);
	}
	
	public boolean isNull() {
		return ValueNull;
	}
	
	public void setNull() {
		ValueNull = true;
		bcdValue = 0;
	}
	
	public void SetAsBcd(double Value) throws DBXException {
		bcdValue = Value;
	}
	
	public double GetAsBcd() throws DBXException {
		return bcdValue;
	}

}

File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TDBXBcdValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXBcdValue.h"


@implementation TDBXBcdValue
-(id)init {
	self =  [super init];
	if (self) {
		[self setDBXType:BcdType];
	}
	return self;
}
-(bool) isNull{
	return valueNull;
}
-(void) SetNull{
	valueNull = YES;
	bcdValue = 0;
}
-(void) SetAsBcd:(double)value{
	bcdValue = value;
}

-(double) GetAsBcd{
	return bcdValue;
}

@end
